app/code/core/Mage/Catalog/Model/Resource/Url.php

```diff
@@ -62,13 +62,26 @@ class Mage_Catalog_Model_Resource_Url extends Mage_Core_Model_Resource_Db_Abstra
      */
     protected $_rootChildrenIds             = [];
 
+    /**
+     * @var int
+     */
+    protected $_productEntityTypeId;
+
+    /**
+     * @var int
+     */
+    protected $_statusAttributeId;
+
     /**
      * Load core Url rewrite model
      *
      */
     protected function _construct()
     {
         $this->_init('core/url_rewrite', 'url_rewrite_id');
+        $this->_productEntityTypeId = Mage::getModel('eav/entity')->setType(Mage_Catalog_Model_Product::ENTITY)->getTypeId();
```

Comment:
how often e.g. during indexing on category view rendering is this resource model created? 
If it's created many times, we need to take a look if this two calls will not degrade performance.

Reply:
I believe the proper method is this:
```suggestion
        $this->_productEntityTypeId = Mage::getSingleton('eav/config')->getEntityType(Mage_Catalog_Model_Product::ENTITY)->getId();
```

Reply:
@tmotyl _construct is execute 2 time in app/code/core/Mage/Catalog/Model/Indexer/Url.php reindexAll :
https://github.com/OpenMage/magento-lts/blob/1.9.4.x/app/code/core/Mage/Catalog/Model/Indexer/Url.php#L254
https://github.com/OpenMage/magento-lts/blob/1.9.4.x/app/code/core/Mage/Catalog/Model/Indexer/Url.php#L257

@colinmollenhour  i get it from
https://github.com/OpenMage/magento-lts/blob/1.9.4.x/app/code/core/Mage/Adminhtml/controllers/Catalog/Product/AttributeController.php#L57
https://github.com/OpenMage/magento-lts/blob/1.9.4.x/app/code/core/Mage/Catalog/Model/Resource/Product/Attribute/Collection.php#L46
I do not know what is the difference between the method you proposed and the one I used, but if yours is better, please confirm, I will change it.

Reply:
Ahh, looks like `setType` just calls the `eav/config` method, lol:
![image](https://user-images.githubusercontent.com/38738/192613251-bad53434-41ec-40b1-b805-bedcd0df814e.png)

I think it is the preferred method still, it does not instantiate any new models and it is used more heavily in the code base.